.github/workflows/build_sqlite_binaries.yml

```diff
@@ -0,0 +1,107 @@
+name: Build SQLite Binaries
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            target: x86_64-linux-gnu
+            platform_dir: linux-x86_64
+            lib_name: sqlite3.so
+            exe_name: sqlite3
+            extra_flags: "-lpthread -ldl -lm"
+          - os: windows-latest
+            target: x86_64-windows-gnu
+            platform_dir: windows-x86_64
+            lib_name: sqlite3.dll
+            exe_name: sqlite3.exe
+            extra_flags: ""
+          - os: macos-latest
+            target: x86_64-macos-none
+            platform_dir: macos-x86_64
+            lib_name: sqlite3.dylib
+            exe_name: sqlite3
+            extra_flags: "-lpthread -ldl -lm"
+          - os: macos-latest
+            target: aarch64-macos-none
+            platform_dir: macos-aarch64
+            lib_name: sqlite3.dylib
+            exe_name: sqlite3
+            extra_flags: "-lpthread -ldl -lm"
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Set up Zig
+      uses: mlugg/setup-zig@v2
+      with:
+        version: 0.14.1
```

Comment:
Zig - good idea. One problem it is 0.14.1, which is not ideal. I want to depend only on stable products. 

I used [perplexity research](https://www.perplexity.ai/search/what-compiler-is-recommend-for-23Z8NmXLQtyG.s4c7g.zzw) and I think good way forward is to use Clang/LLVM.